src/node.cc

```diff
@@ -2487,46 +2491,100 @@ struct node_module* get_linked_module(const char* name) {
   return mp;
 }
 
-// DLOpen is process.dlopen(module, filename).
+#ifdef __POSIX__
+
+static void* node_dlhandle(void* handle) {
+  return handle;
+}
+
+static void node_dlclose(void* handle) {
+  dlclose(handle);
+}
+
+static void* node_dlopen(const char* filename, int flags, char** dlerrmsg) {
+  // libuv does not support passing flags to dlopen.
+  void* handle = dlopen(filename, flags);
+  if (handle != nullptr)
+    return handle;
+
+  const char* errmsg = dlerror();
+  *dlerrmsg = strdup((errmsg != nullptr) ? errmsg : "no error");
+  return nullptr;
+}
+
+#else  // _WIN32
+
+static void* node_dlhandle(void *handle) {
+  uv_lib_t* lib = reinterpret_cast<uv_lib_t*>(handle);
+  return lib->handle;
+}
+
+static void node_dlclose(void *handle) {
+  uv_lib_t* lib = reinterpret_cast<uv_lib_t*>(handle);
+  uv_dlclose(lib);
+  delete lib;
+}
+
+static void* node_dlopen(const char* filename, int flags, char **dlerrmsg) {
+  uv_lib_t* lib = new uv_lib_t;
+  bool is_dlopen_error = uv_dlopen(filename, lib);
+  if (!is_dlopen_error)
+    return lib;
+
+  *dlerrmsg = strdup(uv_dlerror(lib));
+  uv_dlclose(lib);
+  delete lib;
+  return nullptr;
+}
```

Comment:
Can I suggest turning these functions into a RAII class?  That way the uv_lib_t can simply be a data member in the Windows implementation and no juggling of new/delete calls is necessary.